engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/DefinitionExtractor.scala

```diff
@@ -32,7 +32,10 @@ class DefinitionExtractor[T](methodDefinitionExtractor: MethodDefinitionExtracto
     (obj match {
       //TODO: how validators/editors in NodeConfig should be handled for GenericNodeTransformation/WithExplicitMethodToInvoke?
       case e:GenericNodeTransformation[_] =>
-        val returnType = if (e.nodeDependencies.contains(OutputVariableNameDependency)) Unknown else Typed[Void]
+        val returnType = e match {
```

Comment:
we shouldn't do it here. This should be possible return type of this component - current usage with passing union of key and value types does not make sense here. Let's rather use it in TypesInformation.extract method

Reply:
By component you mean "Component" class? IMO we shouldn't do it too much generic. Take a look that this `possibleVariableClasses` takes no arguments. If we would like to do it correctly, we should probable traverse the whole scenario (not definition) and take a look what classes are returned. Also "codefull" approach of declaring components is deprecated. Isn't it?

Reply:
I moved it to `TypesInformation.extract`.